### PR TITLE
*: enable delimited wire protocol upgrade

### PR DESCRIPTION
**Pull request by corverroos**
Enables delimited versions of all libp2p protocols.

category: feature
ticket: #1885

### CODE REVIEW

File: app/peerinfo/adhoc.go [hunk 1]
```diff
@@ -17,18 +17,14 @@ import (
 // or false if the peer doesn't support the protocol,
 // or an error.
 func DoOnce(ctx context.Context, tcpNode host.Host, peerID peer.ID) (*pbv1.PeerInfo, time.Duration, bool, error) {
-	supported, known := p2p.ProtocolSupported(tcpNode, peerID, protocolID)
```
Comment (corverroos) [inline app/peerinfo/adhoc.go:20]: this is not required anymore since all supported versions support this protocol

File: app/peerinfo/peerinfo.go [hunk 2]
```diff
@@ -152,19 +152,6 @@ func (p *PeerInfo) sendOnce(ctx context.Context, now time.Time) {
 			continue // Do not send to self.
 		}
 
-		supported, known := p2p.ProtocolSupported(p.tcpNode, peerID, protocolID)
```
Comment (corverroos) [inline app/peerinfo/peerinfo.go:155]: same here, all supported versions support this protocol

File: core/parsigex/parsigex.go [hunk 3]
```diff
@@ -54,32 +56,18 @@ type ParSigEx struct {
 	subs       []func(context.Context, core.Duty, core.ParSignedDataSet) error
 }
 
-func (m *ParSigEx) handle(s network.Stream) {
```
Comment (corverroos) [inline core/parsigex/parsigex.go:57]: had to migrate this protocol to use `libp2p.RegisterHandler`

File: app/peerinfo/peerinfo.go [hunk 4]
```diff
@@ -24,11 +24,14 @@ import (
 
 const period = time.Minute
 
-var protocolID protocol.ID = "/charon/peerinfo/1.0.0"
+var (
```
Comment (xenowits) [inline app/peerinfo/peerinfo.go:27]: nit: can make this `const`

**Review by xenowits — APPROVED** [review]

**Pull request merged**